client/src/js/services/DocumentService.js

```diff
@@ -0,0 +1,40 @@
+angular.module('bhima.services')
+.service('DocumentService', DocumentService);
+
+DocumentService.$inject = [ '$http', 'util' ];
+
+function DocumentService($http, util) {
+  var service = this;
+  var baseUrl = '/patients/';
+
+  /** expose the service */
+  service.read = read;
+  service.remove = remove;
+  service.removeAll = removeAll;
+
+  /**
+   * This method returns documents on a patient given the patients UUID.
+   *
+   * @param {String|Null} uuid   The patient's UUID  (could be null)
+   * @return {Object}       Promise object that will return patient details
+   */
+  function read(patientUuid) {
+    if (!patientUuid) { return ; }
+
+    return $http.get(baseUrl.concat(patientUuid).concat('/documents'))
+      .then(util.unwrapHttpResponse);
+  }
+
+  /** delete document */
+  function remove(patientUuid, documentUuid) {
+    return $http.delete(baseUrl.concat(patientUuid).concat('/documents/').concat(documentUuid))
```

Comment:
Did you know that `String.prototype.concat()` takes multiple parameters?  You can change this line of code to be:
```js
return $http.delete(baseUrl.concat(patientUuid, '/documents/', documentUuid))
```

[Here is the documentation](https://developer.mozilla.org/fr/docs/Web/JavaScript/Reference/Objets_globaux/String/concat).

This is much cleaner :)